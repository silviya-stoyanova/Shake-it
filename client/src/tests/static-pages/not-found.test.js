import React from 'react'
import { mount } from 'enzyme'
import NotFound from '../../components/static-pages/not-found'

describe('tests for the component NotFound', () => {
    it('should render correctly', () => {
        const wrapper = mount(<NotFound />)
        expect(wrapper.html()).toMatchSnapshot()
    })

    it('should have a div element with className "not-found-wrapper"', () => {
        const wrapper = mount(<NotFound />)
        expect(wrapper.find('div[className="not-found-wrapper"]').length).toEqual(1)
    })

    it('should have a div element with className "not-found-img"', () => {
        const wrapper = mount(<NotFound />)
        expect(wrapper.find('div[className="not-found-img"]').length).toEqual(1)
    })

    it('should consist of exactly 2 div elements', () => {
        const wrapper = mount(<NotFound />)
        expect(wrapper.find('div').length).toEqual(2)
    })

    it('should consist of exactly 3 span elements', () => {
        const wrapper = mount(<NotFound />)
        expect(wrapper.find('span').length).toEqual(3)
    })

    it('should have one span element with classNames "four-digit" and "left-al" and text "4"', () => {
        const wrapper = mount(<NotFound />)
        expect(wrapper.find('span[className="four-digit left-al"]').text()).toEqual('4')
    })

    it('should have one span element with classNames "four-digit" and "right-al" and text "4"', () => {
        const wrapper = mount(<NotFound />)
        expect(wrapper.find('span[className="four-digit right-al"]').text()).toEqual('4')
    })
})