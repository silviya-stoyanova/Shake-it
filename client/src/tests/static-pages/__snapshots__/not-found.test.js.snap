// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for the component NotFound should render correctly 1`] = `"<div class=\\"not-found-wrapper\\"><div class=\\"not-found-img\\"><span class=\\"four-digit left-al\\">4</span><span class=\\"four-digit right-al\\">4</span></div><span class=\\"not-found-txt\\">Sorry, we could not find what you were looking for :(</span></div>"`;
